### PR TITLE
Support multiple GraphQL schema files

### PR DESCRIPTION
**Pull request by eseliger**
This PR adjusts all configs/generators/linters that we use to support multiple `.graphql` files to be stitched together for the final schema.
This should help a bit with the MASSIVE `schema.graphql` file that has grown to around 10k loc by now, by encapsulating different fields of concern into separate files. Using the `extend` keyword, there is nothing that cannot be expressed with separate files that could have been with a single file, so there are no drawbacks to this method, that I'm aware of.

In a follow-up PR, I will restructure the backend code for this a bit and make it so the enterprise schema is not used in OSS (which gives better introspection, because there is less noise from fields that aren't accessible because OSS) and doesn't require us to implement a `defaultXXResolver`.

### CODE REVIEW

File: .graphqlrc.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+schema:
```
Comment (eseliger) [inline .graphqlrc.yml:1]: v2 config format is not supported by the vscode extension anymore, so I needed to migrate.

File: client/branded/tsconfig.json [hunk 2]
```diff
@@ -10,13 +10,6 @@
     },
     "rootDir": ".",
     "outDir": "./out",
-    "plugins": [
```
Comment (eseliger) [inline client/branded/tsconfig.json:13]: This is provided by the graphql extension already, so these two things were competing

File: package.json [hunk 3]
```diff
@@ -15,9 +15,8 @@
     "prettier-check": "yarn -s run prettier --write=false",
     "all:eslint": "DOCSITE_LIST=\"$(./dev/docsite.sh -config doc/docsite.json ls)\" dev/foreach-ts-project.sh yarn -s run eslint --quiet",
     "all:stylelint": "yarn --cwd client/web run stylelint && yarn --cwd client/shared run stylelint && yarn --cwd client/branded run stylelint && yarn --cwd client/browser run stylelint && yarn --cwd client/wildcard run stylelint",
-    "all:tsgql": "yarn --cwd client/web run tsgql validate -p . --exitOnWarn && yarn --cwd client/shared run tsgql validate -p . --exitOnWarn && yarn --cwd client/browser run tsgql validate -p . --exitOnWarn",
```
Comment (eseliger) [inline package.json:18]: This task isn't required as our type generators already do syntax checking on the gql queries

**Comment by felixfbecker** [comment]
Is this standard, or an extension to the IDL?

**Comment by eseliger** [comment]
It's part of the spec https://spec.graphql.org/June2018/#sec-Object-Extensions

**Comment by felixfbecker** [comment]
YEAH that's exciting. I remember a couple years ago it was non-standard. 🥳

**Review by ryanslade — APPROVED** [review]
Just a note that I actually quite like having only one place where I need to look if I want to see our schema.

Also, will this support jumping between files? At the moment with GoLand I can do the usual "code intelligence" type navigation around the schema file.

**Comment by eseliger** [comment]
> Just a note that I actually quite like having only one place where I need to look if I want to see our schema.

I totally get that, but
- A simple text search through all `*.graphql` files gets all results reliably
- The API console still knows about _all_ fields we expose, so you can always tinker in there or use the API doc sidebar on the right
- The current schema is not really correct for OSS, as many fields and entities aren't _actually_ queryable through the API and the only way to find out is to run a query and see it throw some errors "only in enterprise" when hitting those fields
  - This will enable us to only use the schema extensions that are actually available on that instance, so this can be a fix for that problem
  - It will also make it so we don't need the nasty `defaultResolver` implementations for all enterprise features anymore, as those resolvers would not be queryable at all then

Do you think these points outweigh that concern? 

> Also, will this support jumping between files? At the moment with GoLand I can do the usual "code intelligence" type navigation around the schema file.

I think this depends on the editor, as this is contributed by it. I don't have GoLand installed so I cannot test it there, but for VSCode it still supports hover tooltips on _all_ entities across all schema files, and supports j2d _within the same file_. (So cross file code intel works, j2d not, at least in vscode) - BUT the fact that it can correctly resolve them tells me that the information is actually there, it just doesn't do it _now_ - I expect the extension to be updated eventually to support this. We could possibly also contribute that feature, if it bothers us enough.

**Comment by felixfbecker** [comment]
Re: jump to definition - maybe the best lowest denominator we could have is if we can make jump to definition work in Sourcegraph. Would probably work just with basic code intel. That way every dev has a way to explore the schema with codeintel no matter what editor they use :) 
Unfortunately currently it will return definitions in Go instead of the other GraphQL files (just confirmed, but filed a while ago at https://github.com/sourcegraph/sourcegraph/issues/8177). @sourcegraph/code-intel 

But regardless I think it shouldn't block the PR

**Comment by ryanslade** [comment]
Thanks for the detailed reply 😄 

> Do you think these points outweigh that concern?

I'd say the pros definitely outweigh the cons, I just wanted to make sure to voice my concerns and I'm glad I did since it sounds like the issues are relatively easy to solve.

**Review by unknwon — APPROVED** [review]

**Pull request merged**